pkg/deploys/fuse/deployer.go

```diff
@@ -20,122 +21,128 @@ import (
 )
 
 type FuseDeployer struct {
-	id string
+	k8sClient kubernetes.Interface
+	osClient  *openshift.ClientFactory
 }
 
-func NewDeployer(id string) *FuseDeployer {
-	return &FuseDeployer{id: id}
-}
+func NewDeployer() *FuseDeployer {
+	// Instantiate loader for kubeconfig file.
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
I have moved these here as they were being passed from the server but only used here in the fuse deployer. I am not sure if I should create objects like this though? i.e. panic in a "constructor". I wanted to keep a parameter-less "constructor" for each `Deployer` though

Reply:
Not a fan of this change. It makes it very difficult to unit test for instance. The kubeconfig here has to be loaded from disk and will create a real client targeting a real cluster only.
So i think having the dependency injected is much clearer from an API and testing perspective. What is the reason for wanting an empty constructor?

Reply:
Yeah, it felt wrong and your concerns make sense. I'll revert back to passing in params.
I wanted an empty constructor as I wanted each deployer to be responsible for their own setup. A minor issue though and as I said ill revert